R/bundle.R

```diff
@@ -222,15 +222,28 @@ enforceBundleLimits <- function(appDir, totalSize, totalFiles) {
 bundleApp <- function(appName, appDir, appFiles, appPrimaryDoc, assetTypeName,
                       contentCategory, verbose = FALSE, python = NULL,
                       condaMode = FALSE, forceGenerate = FALSE, quarto = NULL,
-                      isShinyApps = FALSE) {
+                      isShinyApps = FALSE, metadata = list()) {
   logger <- verboseLogger(verbose)
 
+  quartoInfo <- inferQuartoInfo(
+    appDir = appDir,
+    appPrimaryDoc = appPrimaryDoc,
+    quarto = quarto,
+    metadata = NULL
+  )
+  if (is.null(contentCategory) && !is.null(quartoInfo)) {
```

Comment:
We do not infer content category for other content types; why are we treating Quarto differently?

If we continue to infer content category, should we have an `inferContentCategory` function?

Reply:
I was treating Quarto differently because `quarto::quarto_publish_app()` sets `contentCategory`, as does push-button publishing, and I was trying to have the manifests I generated match those. I also could see an argument that… if you're publishing a Quarto website, you shouldn't have to tell `writeManifest` that it's a site, not a document.

I can also see an argument for having `writeManifest` be a little more like the "driving a stick shift" version of deployment, and require explicit arguments as much as possible. I'm unsure how `deployApp` should function relative to that (i.e. mirror `writeManifest` as closely as possible versus be a little more automatic).